include/boost/lockfree/spsc_queue.hpp

```diff
@@ -761,6 +831,41 @@ class spsc_queue:
         return base_type::push(t);
     }
 
+    
+#ifdef BOOST_HAS_RVALUE_REFS
```

Comment:
You probably want to detect the absence of
`BOOST_NO_CXX11_RVALUE_REFERENCES` for push, and the absence of both `BOOST_NO_CXX11_RVALUE_REFERENCES` and `BOOST_NO_CXX11_VARIADIC_TEMPLATES` for emplace.